class.photon.php

```diff
@@ -457,7 +457,8 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 				$image_meta = image_get_intermediate_size( $attachment_id, $size );
 
 				// 'full' is a special case: We need consistent data regardless of the requested size.
-				if ( 'full' == $size ) {
+				// We will also need a specific size if thumbnails are set to have dynamic (0 value) height/width. 
+				if ( 'full' == $size || 0 == $image_meta['height'] || 0 == $image_meta['width'] ) {
```

Comment:
I think these need to be strict `===`. If `! $image_meta`, then `0 == $image_meta['height']` would evaluate true.